test/surface/view_test.exs

```diff
@@ -14,4 +14,21 @@ defmodule Surface.ViewTest do
 
     assert result == "Hello world!\n"
   end
+
+  describe "__mix_recompile__?" do
+    @new_file_path "test/support/view_test/templates/foo/recompilation_test.sface"
+    test "returns true when list of templates from the view changes" do
+      on_exit(fn -> File.rm!(@new_file_path) end)
+
+      refute FooView.SurfaceRecompilationHelper.__mix_recompile__?()
+      File.touch!(@new_file_path)
+      assert FooView.SurfaceRecompilationHelper.__mix_recompile__?()
```

Comment:
These will need to be renamed to `MyAppWeb.FooView` if https://github.com/surface-ui/surface/pull/547 gets merged first.